### PR TITLE
Refactor ingest path

### PR DESCRIPTION
**Pull request by cevian**
This is mostly a renaming/reorg refactor. There should not be any logic changes.

- handler->metric batcher
- insert -> copier
- inserter -> dispatcher

Constants are moved to the appropriate files.

Some dead code is removed.

### CODE REVIEW

**Comment by cevian** [comment]
Yeah I wasn't sure whether dispatcher or router is better. This class, mainly, routes (or dispatches) samples to the correct metric batcher.

File: pkg/pgmodel/ingestor/copier.go [hunk 1]
```diff
@@ -174,7 +82,7 @@ func runInserter(conn pgxconn.PgxConn, in chan copyRequest) {
 	}
 }
 
-func inserterGetBatch(batch []copyRequest, in chan copyRequest) ([]copyRequest, bool) {
+func copierGetBatch(batch []copyRequest, in chan copyRequest) ([]copyRequest, bool) {
```
Comment (Harkishen-Singh) [inline pkg/pgmodel/ingestor/copier.go:85]: Should this be `getBatch()` and in comment for this func, we write the copier stuff.

**Comment by Harkishen-Singh** [comment]
If we want a true refactor, there are many things that can be improved. I think we should do them in this PR, otherwise the impact wouldn't be significant. I will comment on those stuff in the PR and then we can make the code sound more pleasant.

**Review by antekresic — APPROVED** [review]
Looks good, new names look clearer and better describe the responsibilities.

**Review by Harkishen-Singh — CHANGES_REQUESTED** [review]
Two things that required updation in our discussion:
1. Follow go-vet when writing comments for functions. Please make comments in this PR in accordance with that.
2. rename `dispatcher()` => `seriesDispatcher()` and in the comment, lets add that "this directs the incoming series to their respective `metric batchers` where each `metric batcher` corresponds to a single metric".

File: pkg/pgmodel/ingestor/copier.go (hunk 1 above)
Comment (cevian) [reply pkg/pgmodel/ingestor/copier.go:85]: I think current name is more descriptive

File: pkg/pgmodel/ingestor/buffer.go [hunk 2]
```diff
@@ -16,13 +15,27 @@ const (
...
 
+type insertDataTask struct {
+	finished *sync.WaitGroup
+	errChan  chan error
+}
+
+// Report that this task is completed, along with any error that may have
+// occurred. Since this is a back-edge on the goroutine graph, it
+// _must never block_: blocking here will cause deadlocks.
+func (idt *insertDataTask) reportResult(err error) {
```
[the remark this answers is not in the document]
Comment (cevian) [reply pkg/pgmodel/ingestor/buffer.go:29]: I believe reportResult is a better name as it is used in other contexts too

**Comment by cevian** [comment]
@Harkishen-Singh Dispather is a better name than seriesDispatcher since what is being dispatches is samples not series

**Comment by Harkishen-Singh** [comment]
@cevian I still find comments of functions no following the convention.

**Review by Harkishen-Singh — APPROVED** [review]

**Pull request merged**